### ISSUE TITLE
Lower case `u` and `v` should not be allowed in output yaml

### ISSUE THREAD

**Issue by elynnwu** [E3SM-Project/scream#2556]
We found discrepancies between `horiz_winds` and lowercase `u` and `v` when no remapping is performed. This is not an issue if we requested uppercase `U` and `V`, or if we remap both horizontally and vertically.

Here are the 4 experiments done:
1. Remapped horizontally and vertically. Output `horiz_winds` , `u` , and `v` . No difference between `horiz_winds[dim=0]` and `u`
2. Remapped horizontally and vertically. Output `horiz_winds`, `U`,  and `V`. No difference between `horiz_winds[dim=0]` and `U`
3. No remapping, output native grid. Output `horiz_winds`, `u` , and `v` . Difference between `horiz_winds[dim=0]` and `u` observed:
```
np.testing.assert_allclose(f2.horiz_winds.isel(dim2=1,time=0), f2.u.isel(time=0))
...
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 2764800 / 2764800 (100%)
Max absolute difference: 95.25008
Max relative difference: 8619548.
 x: array([[  1.160124,  -0.207151,  -0.787418, ..., -11.84242 , -11.617801,
        -11.110476],
       [  3.189489,   2.263207,   1.753559, ..., -12.107219, -11.874316,...
 y: array([[-29.751326, -27.446182, -25.305601, ...,  -1.167284,  -1.154717,
         -1.114421],
       [-28.3936  , -25.810406, -23.467003, ...,   0.335149,   0.314825,.. 
```
4. No remapping, output native grid. Output `horiz_winds`, `U` , and `V`. No difference between `horiz_winds[dim=0]` and `U`

It is unclear where `u` and `v` came from, atmos driver only declares `U` and `V` from `horiz_winds` ([here](https://github.com/E3SM-Project/scream/blob/master/components/eamxx/src/control/atmosphere_driver.cpp#L793)).

Proposed solution: always request `U` and `V`, do not allow outputting `u` and `v`

**Comment by bartgol** [comment]
@elynnwu did you run this with _and_ without the nudging process?

**Comment by elynnwu** [comment]
This is all without nudging.

**Comment by elynnwu** [comment]
This issue is fixed in PR #2574
